Libraries/Text/Text/RCTTextView.m

```diff
@@ -407,38 +408,68 @@ - (void)handleLongPress:(UILongPressGestureRecognizer *)gesture
 }
 #else // [TODO(macOS GH#774)
 
+- (NSView *)hitTest:(NSPoint)point
+{
+  // We will forward mouse click events to the NSTextView ourselves to prevent NSTextView from swallowing events that may be handled in JS (e.g. long press).
+  NSView *hitView = [super hitTest:point];
+  
+  NSEventType eventType = NSApp.currentEvent.type;
+  BOOL isMouseClickEvent = NSEvent.pressedMouseButtons > 0;
+  BOOL isMouseMoveEventType = eventType == NSEventTypeMouseMoved || eventType == NSEventTypeMouseEntered || eventType == NSEventTypeMouseExited || eventType == NSEventTypeCursorUpdate;
+  BOOL isMouseMoveEvent = !isMouseClickEvent && isMouseMoveEventType;
+  BOOL isTextViewClick = (hitView && hitView == _textView) && !isMouseMoveEvent;
+  
+  return isTextViewClick ? self : hitView;
+}
+
 - (void)rightMouseDown:(NSEvent *)event
 {
-  if (_selectable == NO) {
+
+  if (self.selectable == NO) {
     [super rightMouseDown:event];
     return;
   }
-  NSText *fieldEditor = [self.window fieldEditor:YES forObject:self];
-  NSMenu *fieldEditorMenu = [fieldEditor menuForEvent:event];
-
-  RCTAssert(fieldEditorMenu, @"Unable to obtain fieldEditor's context menu");
 
-  if (fieldEditorMenu) {
-    NSMenu *menu = [[NSMenu alloc] initWithTitle:@""];
+  [[RCTTouchHandler touchHandlerForView:self] cancelTouchWithEvent:event];
+  [_textView rightMouseDown:event];
+}
 
-    for (NSMenuItem *fieldEditorMenuItem in fieldEditorMenu.itemArray) {
-      if (fieldEditorMenuItem.action == @selector(copy:)) {
-        NSMenuItem *item = [fieldEditorMenuItem copy];
+- (void)mouseDown:(NSEvent *)event
+{
+  if (!self.selectable) {
+    [super mouseDown:event];
+    return;
+  }
 
-        item.target = self;
-        [menu addItem:item];
+  // Double/triple-clicks should be forwarded to the NSTextView.
+  BOOL shouldForward = event.clickCount > 1;
 
-        break;
-      }
-    }
+  if (!shouldForward) {
+    // Peek at next event to know if a selection should begin.
+    NSEvent *nextEvent = [self.window nextEventMatchingMask:NSLeftMouseUpMask | NSLeftMouseDraggedMask
+                                                  untilDate:[NSDate distantFuture]
+                                                     inMode:NSEventTrackingRunLoopMode
+                                                    dequeue:NO];
+    shouldForward = nextEvent.type == NSLeftMouseDragged;
```

Comment:
Does this become a blocking call until the user's mouse goes up or drags?

Reply:
explanation from original change

>  This spins this thread's runloop here waiting for one of the events, so other events and scheduled tasks can continue to run from this point (as long as they are allowed in a tracking mode). This how mouse tracking is generally done on macOS. For instance, when we do pass the event down into the NSTextView on line 141 below, it will not return until the mouse is up. Also, the dequeue flag is set to NO here so that the mouse up or mouse drag can be processed afterwards by something else.